File: ReqMaker.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace LVLTool
{
    public class ReqMaker
    {

        public static String GetReq(string directory)
        {
            // config, script, texture, lvl, 
            StringBuilder builder = new StringBuilder();
            builder.Append("ucft\n{\n");

            AddFiles(directory, builder, "config");
            AddFiles(directory, builder, "script");
            AddFiles(directory, builder, "texture");
            AddFiles(directory, builder, "loc");
            AddFiles(directory, builder, "lvl");
            
            builder.Append("\n}");
            return builder.ToString();
        }

        private static void AddFiles(string directory, StringBuilder builder, string type)
        {
            string pattern = "*." + type;
            List<string> files = GetFiles(directory, pattern);
            string file = "";
            int lastSlash = 0;
            if (files.Count > 0)
            {
                builder.Append("\n\tREQN\n\t{\n");
                builder.Append(String.Format("\t\t\"{0}\"\n", type));
                for (int i = 0; i < files.Count; i++)
                {
                    file = files[i].Replace("." + type, "");
                    lastSlash = file.LastIndexOf("\\") + 1;
                    file = file.Substring(lastSlash);
                    builder.Append(String.Format("\t\t\"{0}\"\n", file));
                }
                builder.Append("\t}\n");
            }
        }



        private static List<string> GetFiles(string directory, string pattern)
        {
            string[] files = Directory.GetFiles(directory, pattern, SearchOption.TopDirectoryOnly);
            return new List<string>(files);
        }
    }
}

File: dart_translation/lib/req_maker.dart
/// Generates .req file content for LVL extraction.
/// Translated from C# CoreCode/ReqMaker.cs for Linux/CLI use.

import 'dart:io';

import 'package:dart_translation/lvltool_config.dart';

String getReq(String directory) {
  final builder = StringBuffer();
  builder.writeln('ucft');
  builder.writeln('{');
  _addFiles(directory, builder, 'config');
  _addFiles(directory, builder, 'script');
  _addFiles(directory, builder, 'texture');
  _addFiles(directory, builder, 'loc');
  _addFiles(directory, builder, 'lvl');
  builder.writeln('}');
  return builder.toString();
}

void _addFiles(String directory, StringBuffer builder, String type) {
  final pattern = '*.$type';
  final files = _getFiles(directory, type);
  if (files.isNotEmpty) {
    builder.writeln('\tREQN');
    builder.writeln('\t{');
    builder.writeln('\t\t"$type"');
    for (String file in files) {
      final base = file.replaceAll('.$type', '');
      int lastSlash = base.lastIndexOf('/') + 1;
      if(lastSlash == 0) {
        lastSlash = base.lastIndexOf('\\') + 1;
      }
      final name = lastSlash > 0 ? base.substring(lastSlash) : base;
      builder.writeln('\t\t"$name"');
    }
    builder.writeln('\t}');
  }
}

List<String> _getFiles(String directory, String extension) {
  final dir = Directory(directory);
  if (!dir.existsSync()) return [];
  return dir
      .listSync()
      .where((e) => e.path.endsWith('.$extension'))
      .map((e) => e.path)
      .toList();
}
